src/controller/CHIPDeviceController.cpp

```diff
@@ -1637,43 +1653,44 @@ void DeviceCommissioner::OnDone()
                 {
                     if (features.Has(app::Clusters::NetworkCommissioning::NetworkCommissioningFeature::kWiFiNetworkInterface))
                     {
-                        clusters.wifi = path.mEndpointId;
+                        info.network.wifi = path.mEndpointId;
                     }
                     else if (features.Has(
                                  app::Clusters::NetworkCommissioning::NetworkCommissioningFeature::kThreadNetworkInterface))
                     {
-                        clusters.thread = path.mEndpointId;
+                        info.network.thread = path.mEndpointId;
                     }
                     else if (features.Has(
                                  app::Clusters::NetworkCommissioning::NetworkCommissioningFeature::kEthernetNetworkInterface))
                     {
-                        clusters.eth = path.mEndpointId;
+                        info.network.eth = path.mEndpointId;
                     }
                     else
                     {
                         // TODO: Gross workaround for the empty feature map on all clusters. Remove.
-                        if (clusters.thread == kInvalidEndpointId)
+                        if (info.network.thread == kInvalidEndpointId)
                         {
-                            clusters.thread = path.mEndpointId;
+                            info.network.thread = path.mEndpointId;
                         }
-                        if (clusters.wifi == kInvalidEndpointId)
+                        if (info.network.wifi == kInvalidEndpointId)
                         {
-                            clusters.wifi = path.mEndpointId;
+                            info.network.wifi = path.mEndpointId;
                         }
                     }
                 }
             }
             return CHIP_NO_ERROR;
         });
+    return_err = err == CHIP_NO_ERROR ? return_err : err;
 
-    if (err != CHIP_NO_ERROR)
+    if (return_err != CHIP_NO_ERROR)
     {
-        ChipLogError(Controller, "Error parsing Network commissioning features");
+        ChipLogError(Controller, "Error parsing commissioning information");
     }
     mAttributeCache = nullptr;
     mReadClient     = nullptr;
     CommissioningDelegate::CommissioningReport report;
-    report.Set<NetworkClusters>(clusters);
+    report.Set<ReadCommissioningInfo>(info);
     CommissioningStageComplete(err, report);
```

Comment:
Should this be `return_err`?  Seems odd to just report the last error we got (which might be success), even if something earlier failed...

Reply:
thanks. Changed.